### PR TITLE
Clean up formatting of units in unicode and console formats

### LINKED ISSUE #14409: Unicode and console unit representations sometimes include an extraneous space
### Description

As noted in #14407, for units typset in `unicode` or `console` format, a space is included in front, unlike for regular units, yet it is sometimes omitted if a unit scale factor is present.

### Expected behavior

`unit.to_string(format)` should never start with a space, independent of `format`.

### How to Reproduce

From https://github.com/astropy/astropy/pull/14407/files#r1108987447 and https://github.com/astropy/astropy/pull/14407/files#r1109066798:

```python
import astropy.units as u
print(f'{(u.m**-1):unicode}')
 1
 ─
 m
f"{(u.eV*u.s**2).decompose()}"             # space between scale and unit
'1.60218e-19 kg m2'
f"{(u.eV*u.s**2).decompose():unicode}"     # no space between scale and unit
'1.6021766×10⁻¹⁹m² kg'
f"{(1*u.eV*u.s**2).decompose()}"           # space between value and unit
'1.602176634e-19 kg m2'
f"{(1 * u.eV*u.s**2).decompose():unicode}" # space between value and unit
'1.602176634e-19 m² kg'
```

### Versions

Any astropy really.

### PR DESCRIPTION
**Pull request by mhvk**
### Description

This pull request is to address that in `console` and `unicode` format, spaces were not always done properly: for units typeset as a fraction there was a space in front of the unit even without a scale factor, while for units set on a single line, there was no space between the scale factor and the unit.

(Plus some small bits of refactoring while I was at it.)

Fixes #14409

### CODE REVIEW

File: astropy/units/format/console.py [hunk 1]
```diff
@@ -67,7 +71,7 @@ def to_string(cls, unit, inline=True):
             if unit.scale == 1:
                 s = ""
             else:
-                s = cls.format_exponential_notation(unit.scale)
+                s = cls.format_exponential_notation(unit.scale) + " "
```
Comment (olebole) [inline astropy/units/format/console.py:74]: To be nitpicky here: for a scale-only unit this adds a trailing space:
```python
>>> u.Unit(10).to_string('unicode')
'10 '
```
Probably one again use a "parts" list and `" ".join(parts)` at the end, as in `generic.py` (but just with 1-2 parts)?

**Review by olebole — APPROVED** [review]
I am fie with it; just one minor nitpicky comment!

File: astropy/units/format/console.py (hunk 1 above)
Comment (mhvk) [reply astropy/units/format/console.py]: Good catch! the parts use is a bit tricky given the possible multiple lines so I just added the space only when needed. It turned out for `latex` the same mistake was made, so I corrected that while I was at it.

**Comment by mhvk** [comment]
@nstarman - this PR is already approved by @olebole (minus a change I implemented), but his approval is not (yet!) sufficient for merging. Could you have a look?

**Review by pllim — APPROVED** [review]
I am approving by proxy but I will give nstarman a chance to review.

Thanks!

**Review by nstarman — APPROVED** [review]
All the test files shows the changes are good. I didn't catch any more of @olebole'a subtle bugs... LGTM! Thanks @mhvk

**Pull request merged**